mitiq/calibration/tests/test_calibration.py

```diff
@@ -151,6 +151,47 @@ def test_compute_improvements_modifies_IF():
     assert all(IF is not None for IF in IFs)
 
 
+def test_best_strategy():
+
+    test_strategy_settings = Settings(
+        circuit_types=["ghz", "mirror"],
+        num_qubits=2,
+        circuit_depth=10,
+        strategies=[
+            {
+                "technique": "zne",
+                "scale_noise": fold_global,
+                "factory": RichardsonFactory([1.0, 2.0, 3.0]),
+            },
+            {
+                "technique": "zne",
+                "scale_noise": fold_global,
+                "factory": RichardsonFactory([1.0, 3.0, 5.0]),
+            },
+            {
+                "technique": "zne",
+                "scale_noise": fold_global,
+                "factory": LinearFactory([1.0, 2.0, 3.0]),
+            },
+            {
+                "technique": "zne",
+                "scale_noise": fold_global,
+                "factory": LinearFactory([1.0, 3.0, 5.0]),
+            },
+        ],
+        circuit_seed=1,
+    )
+    cal = Calibrator(execute, test_strategy_settings)
+    cal.run()
+    strategy = cal.best_strategy(cal.results)
+    strategy_params = strategy.as_dict()
+
+    if cal.results[-1]["best_improvement_factor"] > 1:
+        assert strategy_params != {}
+    else:
+        assert strategy_params == {}
```

Comment:
This is a bit hard to understand. 
E.g. Why we have `[-1]` in `cal.results[-1]`?
Moreover we can be more explicit e.g.:
 instead of  `assert strategy_params != {}` we could use `assert strategy.technique.name = "zne"` and
instaed of `assert strategy_params == {}` we could use `assert strategy.technique.name = "raw"` .

Reply:
Moreover, can we also check in this test or in a new one, if the best strategy is _really_ the best strategy?

E.g. we could manually compute the expectation value for all strategies (without the calibrator) and check that the optimal strategy gives indeed the best result. Or something along this line.
If the number of things to manually compute is too large, we can make a new test with just 2 strategies and a single circuit type.

Reply:
`cal.results[-1]` refers to the last entry in the list of results, which is where the average improvement factors (averaged over the circuits) are stored.

I've updated the existing test to `assert strategy.technique.name == "zne"` and `assert strategy.technique.name == "raw"`